### PR TITLE
Adding unit tests for schema compare service

### PR DESCRIPTION
**Pull request by llali**
Adding unit tests for schema compare service

### CODE REVIEW

File: extensions/schema-compare/src/schemaCompareMainWindow.ts [hunk 1]
```diff
@@ -671,7 +671,7 @@ export class SchemaCompareMainWindow {
 					.withAdditionalProperties({
 						'operationId': this.operationId
 					}).send();
-				vscode.window.showErrorMessage(loc.cancelErrorMessage(result.errorMessage));
+				this.apiWrapper.showErrorMessage(loc.cancelErrorMessage(result.errorMessage));
```
Comment (SakshiS-harma) [inline extensions/schema-compare/src/schemaCompareMainWindow.ts:674]: I think there was a plan to move away from ApiWrapper. I know that this isn't part of the PR, but should we write new tests without the use of apiWrapper?

Comment (Charles-Gagnon) [reply extensions/schema-compare/src/schemaCompareMainWindow.ts:674]: Yes please 😄

Comment (llali) [reply extensions/schema-compare/src/schemaCompareMainWindow.ts:674]: sure I can do that.

File: extensions/schema-compare/src/test/schemaCompare.test.ts [hunk 2]
```diff
@@ -85,24 +93,305 @@ describe('SchemaCompareMainWindow.start', function (): void {
...
+
+		let schemaCompareResult = new SchemaCompareMainWindow(service.object, mockExtensionContext.object);
+		await schemaCompareResult.start(undefined);
+
+		schemaCompareResult.sourceEndpointInfo = setDacpacEndpointInfo(mocksource);
+		schemaCompareResult.targetEndpointInfo = setDacpacEndpointInfo(mocktarget);
+		await schemaCompareResult.execute();
+		await schemaCompareResult.publishChanges();
+
+		should(showErrorMessageSpy.calledOnce).be.true();
```
Comment (kisantia) [inline extensions/schema-compare/src/test/schemaCompare.test.ts:132]: can there also be a check that the error message matches the expected text? Also for the other tests that are checking that an error message gets shown

File: extensions/schema-compare/src/test/schemaCompare.test.ts [hunk 3]
```diff
@@ -85,24 +93,305 @@ describe('SchemaCompareMainWindow.start', function (): void {
...
+			row: 0,
+			column: 0,
+			columnName: 1,
+			checked: false
+		});
+
+		should(showWarningMessageStub.notCalled).be.true();
+	});
+
+	it('Should show not error if user does not want to publish', async function (): Promise<void> {
```
Comment (kisantia) [inline extensions/schema-compare/src/test/schemaCompare.test.ts:343]: "Should not show" - show and not are switched

**Review by kisantia — APPROVED** [review]
thanks for adding these tests and removing the apiWrapper!

File: extensions/schema-compare/src/test/schemaCompare.test.ts [hunk 4]
```diff
@@ -28,15 +27,24 @@ before(function (): void {
...
+	sinon.restore();
+});
+
 describe('SchemaCompareMainWindow.start', function (): void {
 	before(() => {
 		mockExtensionContext = TypeMoq.Mock.ofType<vscode.ExtensionContext>();
 		mockExtensionContext.setup(x => x.extensionPath).returns(() => '');
 	});
+
+	this.afterEach(() => {
```
[the remark this answers is not in the document]
Comment (llali) [reply extensions/schema-compare/src/test/schemaCompare.test.ts:40]: I have to add it for each test suite 

---
In reply to: [537749040](https://github.com/microsoft/azuredatastudio/pull/13642#discussion_r537749040) [](ancestors = 537749040)

**Pull request merged**